src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java

```diff
@@ -158,13 +162,34 @@ private static void writeTablePropertiesRows(Sink sink)
         final Map<String, DetailNode> propertiesJavadocs = SiteUtil
                 .getPropertiesJavadocs(properties, currentModuleName, currentModuleFile);
 
-        for (String property : properties) {
+        final Set<String> validProperties = orderProperties(properties);
+
+        for (String property : validProperties) {
             final DetailNode propertyJavadoc = propertiesJavadocs.get(property);
             final DetailNode currentModuleJavadoc = propertiesJavadocs.get(currentModuleName);
             writePropertyRow(sink, property, propertyJavadoc, instance, currentModuleJavadoc);
         }
     }
 
+    /**
+     * Rearrange properties alphabetically with property 'tokens' always last.
+     *
+     * @param parameters the javadoc properties.
+     * @return LinkedHashSet contains valid properties for the javadoc.
+     *
+     */
+    private static Set<String> orderProperties(Set<String> parameters) {
```

Comment:
Does this method take into account take a scenario where tokens doesn't exist in property? For instance, in the below documentation, tokens doesn't exist for the properties section. Or, is it being taken by somewhere else as i haven't followed the chain of execution. 

<img width="938" alt="image" src="https://github.com/checkstyle/checkstyle/assets/95259545/1ace7c36-a126-45b9-a87a-320477106e03">